provider/resource_opensearch_sm_policy.go

```diff
@@ -0,0 +1,291 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
+	"github.com/olivere/elastic/uritemplates"
+
+	elastic7 "github.com/olivere/elastic/v7"
+	elastic6 "gopkg.in/olivere/elastic.v6"
+)
+
+var openSearchSMPolicySchema = map[string]*schema.Schema{
+	"policy_name": {
+		Description: "The name of the SM policy.",
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+	},
+	"body": {
+		Description:      "The policy document.",
+		Type:             schema.TypeString,
+		Required:         true,
+		DiffSuppressFunc: smDiffSuppressPolicy,
+		StateFunc: func(v interface{}) string {
+			json, _ := structure.NormalizeJsonString(v)
+			return json
+		},
+	},
+	"primary_term": {
+		Description: "The primary term of the SM policy version.",
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Computed:    true,
+	},
+	"seq_no": {
+		Description: "The sequence number of the SM policy version.",
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Computed:    true,
+	},
+}
+
+func resourceOpenSearchSMPolicy() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provides an OpenSearch Snapshot Management (SM) policy. Please refer to the OpenSearch SM documentation for details.",
+		Create:      resourceOpensearchSMPolicyCreate,
+		Read:        resourceOpensearchSMPolicyRead,
+		Update:      resourceOpensearchSMPolicyUpdate,
+		Delete:      resourceOpensearchSMPolicyDelete,
+		Schema:      openSearchSMPolicySchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				var err error = d.Set("policy_name", d.Id())
+				if err != nil {
+					return nil, err
+				}
+
+				d.SetId(fmt.Sprintf("%s-sm-policy", d.Id()))
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+	}
+}
+
+func resourceOpensearchSMPolicyCreate(d *schema.ResourceData, m interface{}) error {
+	policyResponse, err := resourceOpensearchPostPutSMPolicy(d, m, "POST")
+
+	if err != nil {
+		log.Printf("[INFO] Failed to create OpensearchPolicy: %+v", err)
+		return err
+	}
+
+	d.SetId(policyResponse.PolicyID)
+	return resourceOpensearchSMPolicyRead(d, m)
+}
+
+func resourceOpensearchSMPolicyRead(d *schema.ResourceData, m interface{}) error {
+	policyResponse, err := resourceOpensearchGetSMPolicy(d.Get("policy_name").(string), m)
+
+	if err != nil {
+		if elastic6.IsNotFound(err) || elastic7.IsNotFound(err) {
```

Comment:
NIT: Please remove `elastic6`, AFAIK all the go files are updated to use `elastic7`. Thanks